apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -429,6 +429,53 @@ export async function getAgentPortfolioSnapshots(
   }
 }
 
+/**
+ * Get agent's ranking in a specific competition
+ * @param agentId Agent ID
+ * @param competitionId Competition ID
+ * @returns Object with rank and totalAgents
+ */
+export async function getAgentCompetitionRanking(
+  agentId: string,
+  competitionId: string,
+): Promise<{ rank: number; totalAgents: number }> {
+  try {
+    // Get all latest portfolio snapshots for the competition
+    const snapshots = await getLatestPortfolioSnapshots(competitionId);
+
+    if (snapshots.length === 0) {
+      return { rank: 0, totalAgents: 0 };
+    }
+
+    // Sort by totalValue descending to determine rankings
+    const sortedSnapshots = snapshots.sort(
+      (a, b) => Number(b.totalValue) - Number(a.totalValue),
+    );
+
+    // Find the agent's position (1-based ranking)
+    const agentIndex = sortedSnapshots.findIndex(
+      (snapshot) => snapshot.agentId === agentId,
+    );
+
+    // If agent not found in snapshots, return 0 rank
+    if (agentIndex === -1) {
+      return { rank: 0, totalAgents: sortedSnapshots.length };
```

Comment:
the "rank 0" is problematic because a typical query is: "give my all agents and sort by rank". rank `0` would come before rank `1` and muddy the results.

so, we should just make it nullable/undefined (whichever makes it easier wrt our return types). the client can define handilng the `undefined / totalAgents` logic.